enhancements/ingress/route-subdomain.md

```diff
@@ -0,0 +1,593 @@
+specify a Route as follows:
+
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: hello-openshift
+  namespace: hello-openshift
+spec:
+  # host left unspecified.
+  subdomain: hello
+  # ...
+```
+
+Because the subdomain field is specified, the API will not set a default value
+for the host field.  Instead, **each** router deployment will add an entry to
+the Route's status to indicate the host name under which that particular router
+deployment exposes the Route:
+
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: hello-openshift
+  namespace: hello-openshift
+spec:
+  # ...
+  subdomain: hello
+  # ...
+status:
+  ingress:
+  - conditions:
+    - lastTransitionTime: # ...
+      status: "True"
+      type: Admitted
+    host: hello.apps-internal.mycluster.com
+    routerCanonicalHostname: router-internal.apps-internal.mycluster.com
+    routerName: internal
+    # ...
+  - conditions:
+    - lastTransitionTime: # ...
+      status: "True"
+      type: Admitted
+    host: hello.apps.mycluster.com
+    routerCanonicalHostname: router-default.apps.mycluster.com
+    routerName: default
+    # ...
+```
+
+#### As an application developer, I want to create a Route and have a default host name set based on the shard that exposes the Route
+
+Users can omit `spec.host` and specify `spec.subdomain`, in which case the API
+will leave `spec.host` unset, and each router deployment that exposes the Route
+will do so using the router deployment's domain and the Route's subdomain.  For
+example, suppose the cluster administrator has created the following
+IngressController:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: shard1
+  namespace: openshift-ingress-operator
+spec:
+  domain: shard1.apps.mycluster.com
+  routeSelector:
+    matchLabels:
+      shard: shard1
+```
+
+The application developer specifies a Route as follows:
+
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  labels:
+    shard: shard1
+  name: hello-openshift
+  namespace: hello-openshift
+spec:
+  spec:
+    # host left unspecified.
+    subdomain: hello
+  # ...
+```
+
+Because the subdomain field is specified, the API will not set a default value
+for the host field.  Instead, the router deployment associated with the "shard1"
+IngressController will add an entry to the Route's status to indicate the host
+name under which the router exposes the Route:
+
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  labels:
+    shard: shard1
+  name: hello-openshift
+  namespace: hello-openshift
+spec:
+  # ...
+  subdomain: hello
+  # ...
+status:
+  ingress:
+  - conditions:
+    - lastTransitionTime: # ...
+      status: "True"
+      type: Admitted
+    host: hello.shard1.apps.mycluster.com
+    routerCanonicalHostname: router-shard1.shard1.apps.mycluster.com
+    routerName: shard1
+    # ...
+```
```

Comment:
Technically console wanted this, we didn't deliver it, and as a result the default console DNS name in openshift ootb sucks (console-openshift-console).  While we can't necessarily change that, the use case of cluster agnostic (i.e. unaware of the subname) naming is valid (i want to create a name within the route shard's scope) as an infrastructure user.